roxie/ccd/ccdprotocol.cpp

```diff
@@ -1654,14 +1673,26 @@ class RoxieSocketWorker : public ProtocolQueryWorker
                 uid = queryPT->queryProp("_TransactionId");
             isBlind = queryPT->getPropBool("@blind", false) || queryPT->getPropBool("_blind", false);
             isDebug = queryPT->getPropBool("@debug");
-            if (queryPT->hasProp("_trace") && inlineTraceHeaders)
+
+            if (inlineHeaders)
             {
-                if (queryPT->hasProp("_trace/traceparent"))
-                    inlineTraceHeaders->setProp("traceparent", queryPT->queryProp("_trace/traceparent"));
-                if (queryPT->hasProp("_trace/Global-Id"))
-                    inlineTraceHeaders->setProp("Global-Id", queryPT->queryProp("_trace/Global-Id"));
-                if (queryPT->hasProp("_trace/Caller-Id"))
-                    inlineTraceHeaders->setProp("Caller-Id", queryPT->queryProp("_trace/Caller-Id"));
+                if (queryPT->hasProp("_trace"))
+                {
+                    if (queryPT->hasProp("_trace/traceparent"))
+                        inlineHeaders->setProp("traceparent", queryPT->queryProp("_trace/traceparent"));
+                    if (queryPT->hasProp("_trace/Global-Id"))
+                        inlineHeaders->setProp("Global-Id", queryPT->queryProp("_trace/Global-Id"));
+                    if (queryPT->hasProp("_trace/Caller-Id"))
+                        inlineHeaders->setProp("caller.id", queryPT->queryProp("_trace/Caller-Id"));
+                }
+
+                if (queryPT->hasProp("_client"))
+                {
+                    if (queryPT->hasProp("_client/X-Forwarded-For"))
+                        inlineHeaders->setProp("X-Forwarded-For", queryPT->queryProp("_client/X-Forwarded-For"));
+                    if (queryPT->hasProp("_client/X-Client-ID"))
```

Comment:
this header is passthrough right now, will appear as a span attribute, but can also be added to the COMPLETE line as supplement to the "peer" entry